spec/e2e/association_dependency_spec.rb

```diff
@@ -0,0 +1,183 @@
+
+    [@woe, @absu].each do |band|
+      @user.bands << band
+      @tour.bands << band
+    end
+  end
+
+  def routing_setup
+    [DependentSpec::Route, DependentSpec::Stop].each(&:delete_all)
+    DependentSpec::CALL_COUNT[:called] = 0
+
+    @route1 = DependentSpec::Route.create(name: 'Primary Route')
+    @route2 = DependentSpec::Route.create(name: 'Secondary Route')
+    @tour.routes << [@route1, @route2]
+
+    @philly = DependentSpec::Stop.create(city: 'Philadelphia')
+    @brooklyn = DependentSpec::Stop.create(city: 'Brooklyn')
+    @nyc = DependentSpec::Stop.create(city: 'Manhattan') # Pro Tip from Chris: No good metal shows happen in Manhattan.
+    @providence = DependentSpec::Stop.create(city: 'Providence')
+    @boston = DependentSpec::Stop.create(city: 'Boston') # Boston is iffy, too.
+
+    # We always play Philly. Great DIY scene. If we can't get Brooklyn or Providence, we can do Manhattan and Boston.
+    @route1.stops << [@philly, @brooklyn, @providence]
+    @route2.stops << [@philly, @nyc, @boston]
+  end
+
+  describe 'Grzesiek is booking a tour for his bands' do
+    before(:all) do
+      initial_setup
+    end
+
+    describe 'its primary route stops at every city except NYC and Boston, secondary route includes NYC/Boston' do
+      before do
+        routing_setup
+      end
+
+      context 'and he destroys a Stop with one of those weird BadModels' do
+        before do
+          bad = DependentSpec::BadModel.create
+          bad.user = @user
+          bad.users << @user
+          @boston.poorly_modeled_thing = bad
+        end
+
+        it 'deletes the BadModel in Cypher and does not kill his user account' do
+          @boston.destroy
+          expect(@user).to be_persisted
+          expect(DependentSpec::BadModel.count).to eq 0
+        end
+      end
+
+      context 'the secondary route is destroyed' do
+        before do
+          expect(@philly).to be_persisted
+          [@nyc, @boston].each { |stop| expect(stop).to be_persisted }
+        end
+
+        it 'destroys @nyc and @boston but not @philly' do
+          expect { @route2.destroy }.not_to raise_error
+          expect(@philly).to be_persisted
+          [@nyc, @boston].each { |stop| expect(stop).not_to be_persisted }
+        end
+
+        it 'destroys the linked comment without everything blowing up' do
+          @boston.comments << DependentSpec::Comment.create(note: 'I really hope we do not have to play Boston.')
+          expect(DependentSpec::Comment.count).to eq 1
+          expect { @route2.destroy }.not_to raise_error
+          expect(DependentSpec::Comment.count).to eq 0
+        end
+      end
+    end
+
+    context 'things are going terribly' do
+      describe 'Grzesiek, in frustration, destroys his account' do
+        it 'destroys all bands, tours, routes, stops, and comments' do
+          expect { @user.destroy }.not_to raise_error
+          expect(DependentSpec::Tour.count).to eq 0
```

Comment:
Yeah, don't mind me... ;p

I think the one advantage that is has is that at a glance you can tell than there aren't subtle differences between the lines because they're all executing that one line in the each.